src/directives/style-map.ts

```diff
@@ -18,6 +18,33 @@ export interface StyleInfo {
   readonly [name: string]: string;
 }
 
+// This shim is used on first render, to generate a string to commit
+// rather than manipulate the style object, to be compatible with SSR.
+class StyleDeclaration {
+  setProperty(name: string, value: string) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (this as any)[name] = value;
+  }
+  removeProperty(name: string) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (this as any)[name] = null;
+  }
+  getValue() {
+    return (Object.getOwnPropertyNames(this).reduce((value, prop) => {
+             // eslint-disable-next-line @typescript-eslint/no-explicit-any
+             const v = (this as any)[prop];
+             if (v === null) {
+               return value;
+             }
+             if (prop.indexOf('-') === -1) {
```

Comment:
Arg, seems like vendor-prefixed properties are an issue. e.g.

`webkitAppearance` -> `-webkit-appearance` (need to add a leading `-`)

Easy to do with a regex since the vendor prefixes are pretty [well known](https://developer.mozilla.org/en-US/docs/Glossary/Vendor_Prefix). But makes me not want to do this unconditionally on first render since it'll slow down client-side perf...

Reply:
Went ahead and added the vendor thing; seems like we just go with this for now and revisit in perf testing.